src/util.cpp

```diff
@@ -927,6 +927,37 @@ std::string ArgsManager::GetChainName() const
     return CBaseChainParams::MAIN;
 }
 
+std::string ArgsManager::trimString(std::string str, const std::string& pattern)
+{
+    std::string::size_type front = str.find_first_not_of(pattern);
+    if (front == std::string::npos)
+        return std::string();
+    std::string::size_type end = str.find_last_not_of(pattern);
+    return str.substr(front, end - front + 1);
+}
+
+std::vector<std::pair<std::string, std::string>> ArgsManager::GetConfigOptions(std::istream& stream)
+{
+    std::vector<std::pair<std::string, std::string>> options;
+    std::string str, prefix;
+    std::string::size_type pos;
+    while (std::getline(stream, str)) {
+        if ((pos = str.find('#')) != std::string::npos)
+            str = str.substr(0, pos);
+        str = trimString(str, " \t\r\n");
+        if (!str.empty()) {
+            if (*str.begin() == '[' && *str.rbegin() == ']')
+                prefix = str.substr(1, str.size() - 2) + '.';
+            else if ((pos = str.find('=')) != std::string::npos) {
+                std::string name = prefix + trimString(str.substr(0, pos), " \t\r\n");
+                std::string value = trimString(str.substr(pos + 1), " \t\r\n");
+                options.push_back(std::pair<std::string, std::string>(name, value));
```

Comment:
Minor: can use `std::make_pair(name, value)` for simplicity.

Reply:
`emplace_back` is more simple.